picard/ui/mainwindow.py

```diff
@@ -900,6 +949,23 @@ def play_file(self):
         for file in files:
             QtGui.QDesktopServices.openUrl(self._openUrl(file.filename))
 
+    def internal_play(self):
+        """Play selected tracks with an internal player"""
+        self.player.stop()
+        playlist = QtMultimedia.QMediaPlaylist(self)
+        playlist.setPlaybackMode(QtMultimedia.QMediaPlaylist.Sequential)
+        playlist.addMedia([QtMultimedia.QMediaContent(self._openUrl(file.filename)) for file in self.tagger.get_files_from_objects(self.selected_objects)])
```

Comment:
What does happen if the mediaplayer doesn't support the type of audio files ?

Reply:
As far as I can tell, the media formats supported depends on the O/S and on the codecs actually installed - and that there is no direct way to query QT to determine what formats are supported on the particular system.

It may well be possible to query the O/S directly to determine this, but adding extensive O/S specific code to achieve this doesn't sound like a great idea to me.

It may be possible to determine after the event that the audio didn't play by e.g. getting the QMediaPlayer Error signal with QMediaPlayer::FormatError passed as the parameter - and it would be possible once you had received this to maintain a list of supported or unsupported media types. Or indeed you could try to play the 1st milisecond of each type of media file you encounter to produce such a list - and you could have a list of default media types that each O/S supports to avoid having to do this for types that are known to be supported. Or you could embed a tiny silent example of each type of supported media in Picard and try to play each when Picard starts. **BUT** whatever way you try to achieve this, it seems to me it will be messy.

So personally not sure what the best approach is - perhaps needs some discussion.

Reply:
Currently it will just not play it. On my test there was an error logged to stderr, something like:

> Warning: "No decoder available for type 'application/x-ape'."

But actually this probably comes from gstreamer. I think some basic error handling should be added as part of this PR to log errors and show them in the status bar by connecting to `self.player.error.connect(self._on_player_error)` and having `_on_player_error` doing something like this:

```python
def _on_player_error(self, error):
    self.set_statusbar_message(self.player.errorString(), echo=log.warning)
```